martha/martha_v3.js

```diff
@@ -626,7 +626,13 @@ async function retrieveFromServers(params) {
                 }
             }
         } catch (error) {
-            // Just logging the error for now. Eventually, we'll want to remove this outer try.
+            if (accessMethodTypeIsS3) {
+                // Throw if this is S3 and we failed to get a signed URL. Martha has no concept of a native S3
+                // path so the caller will not have a fallback means of accessing the object.
+                throw error;
```

Comment:
TOL

It can be hard to track down exactly where errors were thrown, and whether they came from library code or ours. If we have to throw, maybe instantiate a new error with some custom context like we do on line 653 (659)  below?

Reply:
This error will have custom context since it is a re-throw of one of the wrapper `RemoteServerError`s generated on line 600, 613 or 625. The [martha v3 handler](https://github.com/broadinstitute/martha/blob/aaecfb5cd9f41d383823a88133d58ae9086a5c84/martha/martha_v3.js#L722) knows about these wrapper`RemoteServerError`s and will extract the `cause` as a parameter to `logAndSendServerError`. If that code finds a `status` [property on the underlying error](https://github.com/broadinstitute/martha/blob/aaecfb5cd9f41d383823a88133d58ae9086a5c84/common/helpers.js#L468) its value will be assigned into the HTTP response, which of course is important information for callers of Martha to have in deciding whether to retry their requests.